unity/Assets/Scripts/BaseFPSAgentController.cs

```diff
@@ -7405,6 +7405,193 @@ ProceduralAsset asset
+
+            float objectSize = Mathf.Max(bounds.size.x, bounds.size.y, bounds.size.z);
+            float objectRadius = Mathf.Sqrt(bounds.extents.x * bounds.extents.x + bounds.extents.z * bounds.extents.z);
+            float cameraHeight = bounds.size.y * cameraHeightMultiplier; // Adjust for 3/4 view
+
+            RenderTexture renderTexture = new RenderTexture((int)renderResolution.x, (int)renderResolution.y, 24);
+            renderCamera.targetTexture = renderTexture;
+
+            List<byte[]> byte_arrays = new List<byte[]>();
+            foreach (float angle in angles) {
+                // Calculate camera position for 3/4 view
+                float radians = (90f - angle) * Mathf.Deg2Rad;
+                Vector3 cameraPosition = new Vector3(
+                    objectRadius * Mathf.Cos(radians),
+                    bounds.extents.y * cameraHeightMultiplier,
+                    objectRadius * Mathf.Sin(radians)
+                );
+                cameraPosition += bounds.center; // Center on object
+                renderCamera.transform.position = cameraPosition;
+                renderCamera.transform.LookAt(bounds.center);
+
+                float dist = FindClosestDistance(renderCamera, bounds, minSlack: 0.025f);
+
+                if (dist < 0) {
+                    errorMessage = "Could not find a suitable distance to render the object.";
+                    byte_arrays = null;
+                    break;
+                }
+//                Debug.Log($"Computed dist: {dist}");
+
+                cameraPosition = bounds.center + Vector3.Normalize(cameraPosition - bounds.center) * dist;
+                renderCamera.transform.position = cameraPosition;
+
+                // Render
+                yield return new WaitForEndOfFrame(); // Ensure the frame is rendered
+
+                Texture2D renderResult = new Texture2D(renderTexture.width, renderTexture.height);
+                RenderTexture.active = renderTexture;
+                renderResult.ReadPixels(new Rect(0, 0, renderTexture.width, renderTexture.height), 0, 0);
+                renderResult.Apply();
+
+                // Save the image
+                byte[] byteArray = renderResult.EncodeToPNG();
+                byte_arrays.Add(byteArray);
+
+//                string filePath = Path.Combine("/Users/lucaw/tmp/yar", $"render_{angle}.png");
+//                File.WriteAllBytes(filePath, byteArray);
+
+                // Clean up
+                Texture2D.Destroy(renderResult);
+            }
+
+            // Reset the camera parameters
+            renderCamera.transform.position = originalPosition;
+            renderCamera.transform.rotation = originalRotation;
+            renderCamera.targetTexture = originalRenderTexture;
+            RenderTexture.active = null;
+            RenderTexture.Destroy(renderTexture);
+
+            actionFinished(byte_arrays != null, byte_arrays);
+        }
+
+        public void RenderObjectFromAngles(
```

Comment:
In the new system you can just make the return type `IEnumerator` and return a `yield new ActionFinished(...)` at some point and the Coroutine will be started for you! so no need to lauch a Coroutine Manually (which can lead to bugs performance issues)

Reply:
In this case you can also return a call to your other function which is an `IEnumerator` `renderObjectFromAngles`

Reply:
From what I could tell, this ended up being more complicated than just leaving it as it is (C# doesn't seem to have a nice way to yield from another function call). Kept as it is for now.